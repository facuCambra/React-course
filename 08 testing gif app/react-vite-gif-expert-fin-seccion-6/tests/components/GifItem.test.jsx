const { render, screen } = require("@testing-library/react");
const { GifItem } = require("../../src/components");

describe ('Tests in <GifItem />', () => {

    const title = 'Title'
    const url = 'https://react.com/'

    test('should make match with the snapshot', () => { 
        const {container} = render(<GifItem title = {title} url = {url} />)
        expect (container ).toMatchSnapshot();

     });

    test('should show the image whith the URL and ALT ', () => { 
        
        render ( <GifItem  title = {title} url = {url} />)
        screen.debug();   //to show the component
        //expect (screen.getByRole('img').src).toBe(url);
        const {src , alt} = screen.getByRole('img');
        expect(src).toBe(url);
        expect(alt).toBe(alt);
     })

     test('should show the correct title', () => { 
        render ( <GifItem  title = {title} url = {url} />)
        expect ( screen.getByText( title)).toBeTruthy();
      })
});