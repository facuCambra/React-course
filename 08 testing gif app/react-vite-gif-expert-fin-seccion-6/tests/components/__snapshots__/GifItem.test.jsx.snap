// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in <GifItem /> should make match with the snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Title"
      src="https://react.com/"
    />
    <p>
      Title
    </p>
  </div>
</div>
`;
